service/src/query_processor.rs

```diff
@@ -109,26 +110,50 @@ impl QueryProcessor {
             receipt,
         } = query;
 
-        // TODO: Emit IndexerErrorCode::IE031 on error
-        let parsed_receipt: SignedReceipt = serde_json::from_str(&receipt)
-            .map_err(|e| QueryError::Other(anyhow::Error::from(e)))?;
+        let parsed_receipt: SignedReceipt = match serde_json::from_str(&receipt)
+            .map_err(|e| QueryError::Other(anyhow::Error::from(e)))
+        {
+            Ok(r) => r,
+            Err(e) => {
+                IndexerError::new(
+                    IndexerErrorCode::IE031,
+                    Some(IndexerErrorCause::new(
+                        "Failed to parse receipt for a paid query",
+                    )),
+                );
+
+                return Err(e);
```

Comment:
This combination of `QueryError` and `IndexerError` makes sense — we're tracking the indexer error internally but return the query error to the client. 👍🏻